### PR TITLE
Two new ghost roles: Pizzeria and DJ Radio Station

### PR DESCRIPTION
**Pull request by ForrestWick**
## About The Pull Request

Goofball/Blutarch told me to PR this upstream, so here ya'll go. Two terrible (or maybe not) ideas for the price of one.

image

First up is the DJ Hut, a role in which two players (at most) will have a little radio station to themselves to vibe in. Unlike the Comms Agent, their role is to create fun gimmicks for a radio show that they'll broadcast to any z level with a comms relay. Wanna be Space Joe Rogan? Space John Madden? Host a talkshow, do sportscasting, run a podcast, talk about conspiracies, you name it. You get an advanced cam console as well, so you can point out random people and ask for interviews from a distance or make fun of security for slipping on banana peels. On top of this, you get a kitchen and garden, a janitor area, and for the emergency situations you'll have a utility room with some stuff to escape the hut if it becomes uninhabitable due to some spacetider.

image
Next up we have the Pizzeria. A group of idiots is running the place, and mama mia it ain't amore. Or maybe they're competent. Who knows? Either way, the pizzeria has a gigabeacon, teleporter, and a full kitchen with plenty of supplies to make pizzas with. You also get a comms relay to advertise with, as well as pamphlets like the hotel. Make some money before the boss comes back!

## Why It's Good For The Game

New ghost roles that could provide neat RP or something fun and gimmicky for dead folks and the station alike.

## Changelog

:cl:
add: Added two new ghost roles: the pizzeria and the dj radio station, with different roles to fill for observers and bored dead players.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 1]
```diff
@@ -829,6 +829,52 @@
...
+	shoes = /obj/item/clothing/shoes/sneakers/black
+	id = /obj/item/card/id/away
+	implants = list(/obj/item/implant/mindshield, /obj/item/implant/exile/noteleport)
+
+//Radio DJ
+
+/obj/effect/mob_spawn/human/radiohost
+	name = "groovy pod"
+	desc = "A humming cryo pod. The machine is attempting to wake up its occupant. You can vaguely hear music coming from inside."
+	mob_name = "a "
```
Comment (Mothblocks) [inline code/game/objects/structures/ghost_role_spawners.dm:860]: ...a?

Comment (ForrestWick) [reply code/game/objects/structures/ghost_role_spawners.dm:860]: Whoops.

**Comment by Colovorat** [comment]
don't we already have some dj station?

**Comment by ForrestWick** [comment]
> don't we already have some dj station?

It's a ruin which I prolly should've replaced, but I didn't know how well that would fly and it's a config disable away from being made redundant.

**Comment by Zytolg** [comment]
Hello. I am a fan of adding new and exciting ruins to /tgstation/, and have been working on shoring up several of our ruins to bring them up to speed with how we've been mapping our stations in 2021. I think these can be a great addition, however some work needs to be done map-wise to really make them pop. I just want to make clear that I'm not pushing for a remap here. Mapping is an artform that takes into account a wide variety of things, and can be very subjective. Not everything I list may be what you have in mind for your maps, however, I believe that addressing them will serve to seriously strengthen your PR. Lets begin!

So there are a few tools that /tgstation/ provides her mappers with that can really serve to make a boring area an exciting one without modifying the space at all. These are turf_decals, and turfs. Over this past year, /tgstation/ has received some amazing additions to both these categories, and including them will only serve to improve your map. I highly encourage you to experiment around with these. 

turf_decals
Turf Decal Example

turfs
Turf Floor Example


Lower down in priority, but still quite important has to do with player movement. You might have to modify your maps to address some of these, but they shouldn't have to be significant alterations (usually we can solve awkward machine/locker placement via edging or pushing them out of the way by 1 tile). My rule of thumb is that if a specific space feels like it's going to be awkward to move around in, then things should be shuffled around until it doesn't feel like that. 

Edge(ing)CASE
tgstation pizza zone machine palcement 1

Another thing is that while tiny fans may serve as useful tools to make sure that amtos behaves itself, I would seriously encourage you to rethink their use here. I think this space can really benefit from appearing like a more inviting space dock. 

Tinyfans bad

On that topic, you also have this, which is a big no-no.
PIZZARADIO BIG NO
You can either balance air on the freezer turfs with the room, or make a new freezer room. Don't use tinyfans to control your atmos like this. 

You have some repeats of the above listed issues on the DJ Radio Station map. 



Unusable Machine Here Oh No DJRADIO
This stasis bed is blocked and unusable by the water tank

Same with the fridge's in the above room
DJRADIO NoFridgeAccess


I find the window placement here an odd choice. 
WindowRocks

This feels empty to me.
DJRADIO Empty Empty

I think this could stand to benefit from a window here. There should be a way to soundproof it. I think it's just lonely if you can't see your radio partner. 
DJRADIO Sadwall


Finally, I think your ruins could benefit from adding posters and other wall mounts. The walls on both ruins feel really plain, and this would be an easy way to again add some life and character to them without changing the footprint of them too much. 
It's quite a long note, this comment is, however I do think by taking the time to resolve these will make this PR something special. On another note, this is the second time I've written this review since Chrome crashed and I lost it. I believe you can really make this pop, and I'm looking forewards to what you do with it in the coming days.

**Comment by ForrestWick** [comment]
> Hello. I am a fan of adding new and exciting ruins to /tgstation/, and have been working on shoring up several of our ruins to bring them up to speed with how we've been mapping our stations in 2021. I think these can be a great addition, however some work needs to be done map-wise to really make them pop. I just want to make clear that I'm not pushing for a remap here. Mapping is an artform that takes into account a wide variety of things, and can be very subjective. Not everything I list may be what you have in mind for your maps, however, I believe that addressing them will serve to seriously strengthen your PR. Lets begin!
> 
> So there are a few tools that /tgstation/ provides her mappers with that can really serve to make a boring area an exciting one without modifying the space at all. These are turf_decals, and turfs. Over this past year, /tgstation/ has received some amazing additions to both these categories, and including them will only serve to improve your map. I highly encourage you to experiment around with these.
> 
> turf_decals Turf Decal Example
> 
> turfs Turf Floor Example
> 
> Lower down in priority, but still quite important has to do with player movement. You might have to modify your maps to address some of these, but they shouldn't have to be significant alterations (usually we can solve awkward machine/locker placement via edging or pushing them out of the way by 1 tile). My rule of thumb is that if a specific space feels like it's going to be awkward to move around in, then things should be shuffled around until it doesn't feel like that.
> 
> Edge(ing)CASE tgstation pizza zone machine palcement 1
> 
> Another thing is that while tiny fans may serve as useful tools to make sure that amtos behaves itself, I would seriously encourage you to rethink their use here. I think this space can really benefit from appearing like a more inviting space dock.
> 
> Tinyfans bad
> 
> On that topic, you also have this, which is a big no-no. PIZZARADIO BIG NO You can either balance air on the freezer turfs with the room, or make a new freezer room. Don't use tinyfans to control your atmos like this.
> 
> You have some repeats of the above listed issues on the DJ Radio Station map.
> 
> Unusable Machine Here Oh No DJRADIO This stasis bed is blocked and unusable by the water tank
> 
> Same with the fridge's in the above room DJRADIO NoFridgeAccess
> 
> I find the window placement here an odd choice. WindowRocks
> 
> This feels empty to me. DJRADIO Empty Empty
> 
> I think this could stand to benefit from a window here. There should be a way to soundproof it. I think it's just lonely if you can't see your radio partner. DJRADIO Sadwall
> 
> Finally, I think your ruins could benefit from adding posters and other wall mounts. The walls on both ruins feel really plain, and this would be an easy way to again add some life and character to them without changing the footprint of them too much. It's quite a long note, this comment is, however I do think by taking the time to resolve these will make this PR something special. On another note, this is the second time I've written this review since Chrome crashed and I lost it. I believe you can really make this pop, and I'm looking forewards to what you do with it in the coming days.

I know about space economy and decals, but it kinda slipped my mind cause I just wanted to be done with these tbh. I'll go back in and make some edits, but overall I feel like these are fine for the most part outside of feeling kinda empty and a bit cramped in some spots

**Comment by ForrestWick** [comment]
image
Alright, here's the updates to the pizza place. Will update the description to reflect it. I added some posters and decals to the dj place, and a window with a toggle-able curtain between the booths.

**Comment by Ghommie** [comment]
> It's a ruin which I prolly should've replaced, but I didn't know how well that would fly and it's a config disable away from being made redundant.

Well, then go ahead and replace it, just don't use the same names for the map or it'll be disabled.

**Comment by jughu** [comment]
Didnt we remove hotel and space bar cuz people abused the teleporter?

**Comment by Ghommie** [comment]
> 
> 
> Didnt we remove hotel and space bar cuz people abused the teleporter?

Yes, and I was the bartender teleporting on stations at random with the hand tele I rushed before anyone because it honestly felt funny back in my newbie days.

Also that's probably a rhetorical question. You were the one that removed the space bar (#24707).

**Comment by jughu** [comment]
Can we mqke it so peop p e who have a exile implqnt cant use teleporters?

**Comment by ForrestWick** [comment]
> Can we mqke it so peop p e who have a exile implqnt cant use teleporters?

Literally look at the code in this pr

**Comment by MessoDepresso** [comment]
> > Can we mqke it so peop p e who have a exile implqnt cant use teleporters?
> 
> Literally look at the code in this pr

To put it more politely, @jughu, the roles already spawn with stuff that disables teleporting.

Also, @ForrestWick, I'm curious - How will the pizza place advertise? Do you have an equivalent to the hotel's brochures that they can toss through to show theyre open for business?
Also your pictures are out of order :b

**Comment by ForrestWick** [comment]
> > > Can we mqke it so peop p e who have a exile implqnt cant use teleporters?
> > 
> > Literally look at the code in this pr
> 
> To put it more politely, @jughu, the roles already spawn with stuff that disables teleporting.
> 
> ~~Also, @ForrestWick, I'm curious - How will the pizza place advertise? Do you have an equivalent to the hotel's brochures that they can toss through to show theyre open for business?~~
> ((Disregard, I just read the part about this))
> 
> Also your pictures are out of order :b

Agh. I'll fix it later then. Not a high priority tbh

**Comment by ATH1909** [comment]
does the pizzeria have, uh, an actual way to receive payment for the food they teleport to the station upon request? if not, are you expecting people to break into the teleporter room (opening them up to being validhunted by sec) in order to visit them?

**Comment by ATH1909** [comment]
do the radio hosts have a way to get bigtext or otherwise make themselves stand out?

the pizzeria lads would probably also appreciate a megaphone or something to advertise their place with, actually.

**Comment by ForrestWick** [comment]
> does the pizzeria have, uh, an actual way to receive payment for the food they teleport to the station upon request? if not, are you expecting people to break into the teleporter room (opening them up to being validhunted by sec) in order to visit them?

Hm. This is a good point I suppose, and something I didnt think of. I'm not in the tg discord but i know Goof is in skyrat so I'll ask him about this to see if we can figure something out.

**Comment by Bawhoppen** [comment]
In my opinion, these don't really add that much, and may likely in some ways hurt the game. They aren't particularly fleshed out, and just reuse the same ability that an assistant in maintenance could do, while getting rid of the ingenuity when an enterprising player does it on their own creative ambition. 

From my experience in the game, I also predict that after the first few weeks of them being new, people will rarely use the spawners as well. Comms agents and the ashlizards are popular because they're semi-antags (and also a tiny bit rare-ish to keep them fresh). These don't really add anything unique for player interest, other than for RP. And although I'm all for RP, I don't suspect that these would be more popular than simply being a creative player who opens their own pizza business, or the ambitious assistant who finds the existing Ruskie DJ Station ruin.
Because the ruins will have standardized those practices as well, the excitement and uniqueness of people going out of their way to do that will also be gone. 
This could be mitigated if there was some special quality that really fleshed out these new ruins, but it just seems to be things that can exist otherwise. Having not any special development behind them just emphasizes the previous issues I listed.
Beyond all that, I also note that these may just divide the round's player population more as well, taking more people offstation and doing largely their own thing rather than contributing to the round. Ghost roles have been significantly scrutinized for that reason already, and where the pseudo-antags like ashlizard and comms agent impact the round in a positive way (by adding light conflict), these don't add that much beyond fluff or for RP. And like I said before, I believe that the RP aspect of it would be better handled organically and by creative players.

Additionally to everything else, there is of course also the potential for abuse that has already been mentioned as well. While not a total roadblock, it's worth thinking about. Ashlizards and comms agents have some pretty big rules and obstacles to prevent them from doing things they shouldn't. Having a straight-shot teleporter though is a lot more risky for players looking to abuse the system. Personally, I don't mind having the flexibility and risk of rule-breaking, but it's worth considering either way.

Anyways, I hope I didn't sound too harsh. Nothing against you, but just don't feel this specific contribution fits well on tg. Hope you aren't discouraged from more upstream involvement though! Would love to have you around.

**Comment by Maurukas** [comment]
Barring a bit of excess loot, the maps themselves are quite good. A revamped DJ station, and a fully featured radio host role is something that's worked very well in other codebases, and something I would like to see on tg.

The real issue here is the ghost roles. There have been quite a few issues with ghost roles current and past. In particular, ghost roles that are designed to interact with the station tend to get treated as free respawns to return to the station, with all the administrative headaches that come with this. The pizerria role is almost identical to one that was removed from tg five years ago for being a problem. Both the roles are open ended - they have no real goals, and no real progression. There's nothing to tie players to the role except vacuum, no unique mechanics for them to explore, and no penalties if they ignore those roles.

I think the most likely result of merging these will be the administration disabling these ruins in config. So I think this PR is better suited to a downstream, where the culture, style of gameplay and the admins better fit this kind of open ended, unstructured role.

Please don't be discouraged. This mapping is quite good, and we are actively looking for more ruins - both space based and especially on Icemoon. I would love to see contributions from you in the future.

**Pull request closed without merging**